### ISSUE TITLE
Cypress performance degrades to the point of failure in Chrome 79 on Ubuntu

### ISSUE THREAD

**Issue by c32hedge** [cypress-io/cypress#6023]
### Current behavior:

I can run my tests fine using Cypress 3.4.1 or 3.8.0 against Chrome 78 on Ubuntu 18.04, but if I update to Chrome 79, Cypress starts hanging in random places in my test (as in, the tests hang 0-20 seconds in, but at a different point each time I try). All the Cypress controls (e.g. Stop button) become unresponsive and I can only close the Chrome window.

### Desired behavior:

I want to be able to use Cypress against the latest Chrome version.

### Steps to reproduce: (app code and test code)

I've reproduced this using the following basic test:
```js
describe('page', () => {
  for (let i=1; i <= 30; i++) {
    it(`test ${i}`, () => {
      cy.visit('https://www.google.com');
    });
  }
});
```
It's usually hanging indefinitely after about 12-15 tests when using `cypress open`. Note that it's very choppy even when the tests are still running, with multiple seconds between updates to the elapsed time and the rest of the window.

Interestingly, `cypress run --browser "chrome"` does better. It's still choppy, but it eventually finishes even if I bump `i` up to 100.

If I uninstall Chrome 79 and install Chrome 78 from an archive, the tests run to completion and the window remains responsive for the whole run.

### Versions

Cypress 3.4.1 - 3.8.0 (at least--I didn't try anything older)
Chrome Version 79.0.3945.88 (Official Build) (64-bit)
Ubuntu 18.04.3 LTS

This is **not** seen in Chrome 79 on Mac or Windows

**Comment by c32hedge** [comment]
Of note, a coworker is using Chrome 79 on Mac and does not have this issue.

**Comment by c32hedge** [comment]
It is also working fine on Windows.

**Comment by WTK** [comment]
I've also experienced this issue, same browser, same OS

**Comment by jennifer-shehane** [comment]
I'm not experiencing any slowdown in MacOS. Will have to have someone on the team run from ubuntu.

**Comment by sainthkh** [comment]
I've tested it with my local Ubuntu. But I cannot reproduce it. But can you test this code?

```js
Cypress.Commands.add('componentHasLoaded', () => {
  cy.get('#storybook-preview-iframe').then(($iframe) => {
    const doc = $iframe.contents()

    cy.wrap(doc.find('#root')).should('not.be.empty')
  })
})

describe('issue #5987', () => {
  it('test', () => {
    cy.visit('https://unruffled-lalande-8277bd.netlify.com/?path=/story/button-0--default-1')
    cy.get('#explorerbutton-0').click()
    cy.get('[id^=explorer]').each(($component) => {
      cy.wrap($component)
      .click()
      .get(`[id^=${$component[0].id}--]`)
      .each(($story) => {
        cy.wrap($story)
        .click()
        .wait(50)
        .componentHasLoaded()
      })
    })
  })
})
```

This is the test code I've got from #5987. When the test goes to about number 30, everything slows down significantly. Is this the situation you're talking about?

**Comment by c32hedge** [comment]
@sainthkh  I see the performance degradation and hanging I described above in a very basic scenario (tests with nothing but a `cy.visit`) and only in Chrome 79.

Running your example, I see the slowdown you describe in Chrome 78. If I run it in Chrome 79, it stops and hangs indefinitely somewhere around `Button-9` through `Button-15`, so I believe these are separate issues.

@jennifer-shehane Yes, someone else on our team is using Chrome 79 on a Mac without issue. We are only seeing this on Ubuntu.

**Comment by bapohka** [comment]
such problem on cypress/browsers:node12.6.0-chrome75

**Comment by clarmso** [comment]
Has anyone tried to run the tests with the current google-chrome-unstable (currently Chrome 81)?  I got some slowness and freezing on Cypress 3.6.0-3.8.1 with Chrome 79 as well but Cypress with Chrome 81 runs well for me so far on Ubuntu 16.04 LTS.

TL;DR: My environment
Cypress 3.6.1 and 3.8.1
Chrome 81.0.4000.3 (Official Build) dev (64-bit)
Ubuntu 16.04 LTS

**Comment by clarmso** [comment]
@jennifer-shehane I got some performance issues on Chrome 79+ with Cypress 3.6.1 and 3.8.1 on Ubuntu, too. Setting DEBUG=cypress:* on the CLI generates tons of logs.  Are there any particular items on the log that may indicate performance issues in this case?

**Comment by c32hedge** [comment]
@jennifer-shehane anything else we can do to move the investigation forward? Based on the activity it seems this is impacting a number of users.

**Comment by clarmso** [comment]
Looks like the performance is specific to Chrome 79, not Cypress. :( https://support.google.com/chrome/thread/24842699?hl=en

**Comment by jennifer-shehane** [comment]
In order to narrow this down to Chrome 79 specifically, when running your tests select "Electron 78" during `cypress open`, this will not use Chrome 79 and would isolate the cause of the issue to Chrome. 

![Screen Shot 2020-01-14 at 12 11 04 PM](https://user-images.githubusercontent.com/1271364/72317235-f6c8ca00-36c6-11ea-9745-9695ffa672a6.png)

To take an extra step in isolating it to the Chrome version (and not our Electron install), you can download the last stable Chrome 78 version from this url https://chromium.cypress.io/ and see if the freezing can be isolated to a specific Chrome version.

<img width="887" alt="Screen Shot 2020-01-14 at 12 10 01 PM" src="https://user-images.githubusercontent.com/1271364/72317215-ea447180-36c6-11ea-9f5d-efac7345150b.png">

**Comment by c32hedge** [comment]
I will try with Electron. I did go back and forth between Chrome 78 and 79 several times: the problem would disappear after downgrading to 78, and reappear after installing 79.

**Comment by jennifer-shehane** [comment]
There is an open Chromium bug involving freezing in Linux here: https://bugs.chromium.org/p/chromium/issues/detail?id=1036427

There is a more generic bug involving freezing on non-Linux machines here: https://bugs.chromium.org/p/chromium/issues/detail?id=1036742

For now we're treating this as a bug in Chrome. 

Please open a new issue if you're 'freezing'/performance is affected also on Chrome browser <79. Follow the instructions [here](https://github.com/cypress-io/cypress/issues/6023#issuecomment-574011973) to determine if the performance degrades on Chrome browsers <79.

**Comment by c32hedge** [comment]
@jennifer-shehane  looking through the comments on Chromium issue 1036427, it looks like they're pointing to https://bugs.chromium.org/p/chromium/issues/detail?id=1036742 for the linux-specific freezing.

**Comment by c32hedge** [comment]
**Electron 78 does not hang for me**. I also tried Chromium 79--it wouldn't even open the browser and failed with an error about failing to connect to the DevTools protocol:

> Cypress failed to make a connection to the Chrome DevTools Protocol after retrying for 20 seconds.
> 
> This usually indicates there was a problem opening the Chrome browser.
> 
> The CDP port requested was 40845.
> 
> Error details:
> 
> Error: connect ECONNREFUSED 127.0.0.1:40845 at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1056:14)

**Comment by jennifer-shehane** [comment]
Fixed the links, thanks. 

Most of the CDP connection issues have been resolved in our latest 3.8.2 release, are you using the latest release? This error will occur if you have 2 Cypress applications open https://github.com/cypress-io/cypress/issues/5613 or if you are directly connecting to the Chrome debugger protocol by some other means in your project.

**Comment by clarmso** [comment]
Chrome 80 desktop has been promoted to stable. I just tried it out and it's definitely an improvement from the intermittent freezing we've encountered on Chrome 79. @c32hedge Would you please try Chrome 80 with Cypress?

https://chromereleases.googleblog.com/2020/02/stable-channel-update-for-desktop.html

**Comment by c32hedge** [comment]
I'm able to use Chrome 80 on Ubuntu without noticing any lagging or freezing.

**Comment by jennifer-shehane** [comment]
We've released a docker image for Chrome 80 found here: https://github.com/cypress-io/cypress-docker-images/tree/master/browsers/node13.6.0-chrome80-ff72 at `cypress/browsers:node13.6.0-chrome80-ff72`. 

We recommend that if you were encountering this performance issue in Chrome 79 to update to using Chrome 80. Chrome will not retroactively fix the issue for Chrome 79, so this is the only fix available. 

Closing this as resolved as this was an external Chrome bug. 

- If upgrading to Chrome 80 does not fix your performance issue, please comment in this issue.
- If you're encountering a bug similar to this in the future - please open a new issue with a reproducible example.

**Comment by mdkhwajams** [comment]
I am experiencing the same freezing issue with **cypress/browsers:node12.16.1-chrome80-ff73** I believe it uses chrome 80

**Comment by mdkhwajams** [comment]
> I am experiencing the same freezing issue with **cypress/browsers:node12.16.1-chrome80-ff73** I believe it uses chrome 80

I fixed the issue by breaking down the spec files into smaller tests.

**Comment by lounsbrough** [comment]
FWIW - we were also having this issue on `node12.16.1-chrome80-ff73`. Resolved by moving to `node12.19.0-chrome86-ff82`.
